### PR TITLE
[openxr] Cleanup code gen

### PR DESCRIPTION
**Pull request by MarkY-LunarG**
Remove dependencies between DX  and Vulkan codegen.  Remove multiple re definitions of similar variables.  Reorganize files.

### CODE REVIEW

**Comment by MarkY-LunarG** [comment]
Should resolve #1725.

File: framework/generated/base_generators/base_struct_decoders_header_generator.py [hunk 1]
```diff
@@ -36,6 +35,11 @@ def generate_struct_info(self):
         """Performs C++ code generation for the feature."""
         body = '\n'
         for struct in self.all_structs:
+            if struct in self.all_struct_aliases or struct in self.all_unions or self.is_struct_black_listed(
```
Comment (jzulauf-lunarg) [inline framework/generated/base_generators/base_struct_decoders_header_generator.py:38]: More DRY.  Have seen this multipart clause before should be a function, especially where it's new.

**Review by jzulauf-lunarg — COMMENTED** [review]
1. Need to clobber all unneeded wrap and formatting changes.  Just too noisy.  There's a lot here, don't add more.
2. All renaming (and wrap/whitespace) should be in separate commits labeled renaming/whitespace only
3. More commits at this stage.  For example. Split off DX -> standalone, Vk -> Khronos, then Oxr -> khronos, then cleanup dead code.  So much going on at once, impossible to follow it.
4. Could really use some documentation about what is going on in the cleanup.  Design goal, changes to class hierarchy, changes to generation flow
5. Since we're renaming and generating new files lets give all the python files unique names.  (these name changes should be in their own commit.
6. We're going to want broad review of these changes by the rest of the team

File: framework/generated/khronos_generators/khronos_struct_handle_mappers_header_generator.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+import sys
+from base_generator_defines import write
+
+
+class KhronosStructHandleMappersHeaderGenerator():
+    """Base class for generating struct handle mappers header code."""
+
+    def endFile(self):
+        platform_type = self.get_api_prefix()
```
[the remark this answers is not in the document]
Comment (jzulauf-lunarg) [reply framework/generated/khronos_generators/khronos_struct_handle_mappers_header_generator.py:32]: or... (see next comment)

**Comment by MarkY-LunarG** [comment]
Closing because we're merging directly into dev little by little.

**Pull request closed without merging**